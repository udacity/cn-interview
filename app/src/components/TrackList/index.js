import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import { TRACK_ARRAY } from '../../utils/helper';

export default class TrackList extends Component {
  static propTypes = {
    onSelect: PropTypes.func,
    selected: PropTypes.string,
  };

  _Item = (track, selected, onSelect) => {
    const className = classNames('radio-item', {
      selected: track.name === selected,
    });
    return (
      <Link to={`/${track.router}`} style={{ textDecoration: 'none' }}>
        <div className={className} onClick={() => onSelect(track.name)}>
          <h3 className="h-title h-radio">{track.name}</h3>
        </div>
      </Link>
    );
  };

  render() {
    const { onSelect, selected } = this.props;
    const tracks = [...TRACK_ARRAY, { name: '全部课程', router: 'all' }];
    return (
      <ul className="no-list-style">
        {tracks.map(track => (
          <li key={track.name}>{this._Item(track, selected, onSelect)}</li>
        ))}
      </ul>
    );
  }
}