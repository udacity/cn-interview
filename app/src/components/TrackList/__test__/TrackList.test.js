import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import TrackList from '..';

it('renders correctly', () => {
  const wrapper = shallow(<TrackList selected="全部课程" />);
  expect(wrapper).toMatchSnapshot();
});

it('selected right item ', () => {
  const wrapper = mount(
    <MemoryRouter>
      <TrackList selected="全部课程" />
    </MemoryRouter>
  );
  expect(
    wrapper
      .find('.radio-item')
      .slice(-1)
      .hasClass('selected')
  ).toBe(true);
});

it('can trigger onSelect prop', () => {
  const handleSelect = jest.fn();
  const wrapper = mount(
    <MemoryRouter>
      <TrackList selected="全部课程" onSelect={handleSelect}/>
    </MemoryRouter>
  );
  wrapper
    .find('.radio-item')
    .at(1)
    .simulate('click');
  expect(handleSelect).toHaveBeenCalled();
});
