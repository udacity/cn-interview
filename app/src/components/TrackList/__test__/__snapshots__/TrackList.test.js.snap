// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="no-list-style"
>
  <li
    key="数据科学"
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/data-science"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          数据科学
        </h3>
      </div>
    </Link>
  </li>
  <li
    key="Android"
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/android"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          Android
        </h3>
      </div>
    </Link>
  </li>
  <li
    key="网站开发"
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/web-development"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          网站开发
        </h3>
      </div>
    </Link>
  </li>
  <li
    key="软件工程"
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/software-engineering"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          软件工程
        </h3>
      </div>
    </Link>
  </li>
  <li
    key="iOS"
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/ios"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          iOS
        </h3>
      </div>
    </Link>
  </li>
  <li
    key="佐治亚理工学院计算机科学硕士"
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/georgia-tech-masters-in-cs"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          佐治亚理工学院计算机科学硕士
        </h3>
      </div>
    </Link>
  </li>
  <li
    key="非技术类"
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/non-tech"
    >
      <div
        className="radio-item"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          非技术类
        </h3>
      </div>
    </Link>
  </li>
  <li
    key="全部课程"
  >
    <Link
      replace={false}
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/all"
    >
      <div
        className="radio-item selected"
        onClick={[Function]}
      >
        <h3
          className="h-title h-radio"
        >
          全部课程
        </h3>
      </div>
    </Link>
  </li>
</ul>
`;
